test/Microsoft.DotNet.ShellShim.Tests/ShellShimRepositoryTests.cs

```diff
@@ -45,6 +45,27 @@ public void GivenAnExecutablePathItCanGenerateShimFile()
             stdOut.Should().Contain("Hello World");
         }
 
+        // Reproduce https://github.com/dotnet/cli/issues/9319
```

Comment:
@peterhuene @nguerrera added test to repro. I reverted 2 commits, see the repro and re apply the 2 commits back, and see it is fixed